python_modules/dagster/dagster/_core/storage/asset_check_execution_record.py

```diff
@@ -37,6 +42,7 @@ class AssetCheckExecutionRecord(
     NamedTuple(
         "_AssetCheckExecutionRecord",
         [
+            ("key", AssetCheckKey),
```

Comment:
The existing database methods return a mapping from key to record, which is why we have gotten away without this in the past.

However, this doesn't play nicely with the InstanceLoadableBy scheme, as this just returns a flat list of objects, so it's nice to have the key available on each of those objects.

Reply:
nit: `@record` while you are here to tidy the class up